File: lesson_secure_coding/py/injection_problem.py
import os
import sqlite3

filename = 'example.py.db'

def reset_database():
    try:
        os.remove(filename)
    except OSError:
        pass

def setup_database():
    conn = sqlite3.connect(filename)
    cursor = conn.cursor()
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS users (
            id INTEGER PRIMARY KEY,
            name TEXT
        )
    ''')
    cursor.execute('INSERT INTO users (name) VALUES ("Alice")')
    cursor.execute('INSERT INTO users (name) VALUES ("Bob")')
    cursor.execute('INSERT INTO users (name) VALUES ("Cartman")')
    conn.commit()
    conn.close()

def get_user(user_id):
    conn = sqlite3.connect(filename)
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE id = {user_id}"  # Vulnerable to SQL Injection
    print(f"executing query: {query}")
    cursor.execute(query)
    user = cursor.fetchall()
    conn.close()
    return user

reset_database()
setup_database()
user = get_user("1 OR 1=1")

print(user)

File: lesson_secure_coding/node/injection_problem.js
const sqlite3 = require('sqlite3').verbose();

let database = new sqlite3.Database(':memory:');
let userInput = "Bob' OR '1'='1"

function initDb(db) {
  db.run("CREATE TABLE users (id INTEGER PRIMARY KEY, name TEXT, email TEXT)");
  db.run("INSERT INTO users (name, email) VALUES ('Alice', 'alice@example.com')");
  db.run("INSERT INTO users (name, email) VALUES ('Bob', 'bob@example.com')");
  console.log("Database initialized with sample data.");
}

database.serialize(() => {
  initDb(database);
  let query = `SELECT * FROM users WHERE name = '${userInput}'`;
  console.log(`Executing query: ${query}`);
  database.all(query, [], (err, rows) => {
    rows.forEach((row) => {
      console.log(row);
    });
  });
});

database.close();
